### ISSUE TITLE
Unpaired Image-to-Image Translation using Cycle-Consistent Adversarial Networks

### ISSUE THREAD

**Issue by leo-p** [leo-p/papers#30]
https://arxiv.org/pdf/1703.10593.pdf

Image-to-image translation is a class of vision and graphics problems where the goal is to learn the mapping between an input image and an output image using a training set of aligned image pairs. However, for many tasks, paired training data will not be available. We present an approach for learning to translate an image from a source domain X to a target domain Y in the absence of paired examples. Our goal is to learn a mapping G : X → Y such that the distribution of images from G(X) is indistinguishable from the distribution Y using an adversarial loss. Because this mapping is highly under-constrained, we couple it with an inverse mapping F : Y → X and introduce a cycle consistency loss to push F(G(X)) ≈ X (and vice versa). Qualitative results are presented on several tasks where paired training data does not exist, including collection style transfer, object transfiguration, season transfer, and photo enhancement, etc. Quantitative comparisons against several prior methods demonstrate the superiority of our approach.

**Comment by leo-p** [comment]
## Summary:
- *Dataset:* [Cityscapes](https://www.cityscapes-dataset.com/), [CMP Facade](http://cmp.felk.cvut.cz/~tylecr1/facade/), [UT Zappos50k](http://vision.cs.utexas.edu/projects/finegrained/utzap50k/) and [ImageNet](http://www.image-net.org/).
- *Objective:* Image-to-image translation to perform visual attribute transfer using unpaired images.
- *Code:* [CycleGAN](https://github.com/junyanz/CycleGAN)

## Inner-workings:
Basically two GANs for each domain with their respective Generator and Discriminator plus two additional losses (called consistency losses) to make sure that translating to the other domain then back yields an image that is still realistic.
<img width="1030" alt="screen shot 2017-06-02 at 10 24 45 am" src="https://cloud.githubusercontent.com/assets/17261080/26717449/bcd8a9cc-477d-11e7-9137-fd277a0ec04f.png">

For the consistency los they use a pixel-wise L1 norm:
<img width="371" alt="screen shot 2017-06-02 at 10 31 22 am" src="https://cloud.githubusercontent.com/assets/17261080/26717733/bc088cdc-477e-11e7-96af-2defa06a1660.png">

## Architecture:
Based on [Perceptual losses for real-time style transfer and super-resolution](https://arxiv.org/pdf/1603.08155.pdf), code available [here](https://github.com/jcjohnson/fast-neural-style).
Training seems to employ several tricks and then even use a batch of 1.

## Results:
Very impressive and the really key point is that you don't need paired images which makes this trainable on any domain with the same representation behind.
<img width="1020" alt="screen shot 2017-06-02 at 10 26 29 am" src="https://cloud.githubusercontent.com/assets/17261080/26717502/f6d1fb7e-477d-11e7-8174-7bdd621cf1b6.png">
